### PR TITLE
[diffusion] hardware: SiluAndMul/RMSNorm/LayerNorm MUSA implementations (custom ops, 12/N)

### PR DESCRIPTION
**Pull request by yeahdongcn**
## Motivation

This PR continues the ongoing effort (tracked in https://github.com/sgl-project/sglang/issues/16565) to add MUSA-specific implementations of `SiluAndMul` and `RMSNorm` via `forward_musa` dispatch.

It leverages the `fused_add_rmsnorm` kernel introduced in https://github.com/sgl-project/sglang/pull/17946. Since this PR depends on that change, it is currently marked as **Draft**.

## Modifications

* Implement `forward_musa` for the `SiluAndMul` and `RMSNorm` custom ops.
* Integrate the fused `add + RMSNorm` path when residual is provided.
* Ensure correct handling of non-contiguous inputs and dtype preservation.

#### Functional Tests

* **TestSiluAndMul**: 31/31 passed

  * Covers 2D/3D shapes
  * fp16 / bf16 / fp32
  * Edge cases

* **TestRMSNorm**: 61/61 passed

  * With and without residual
  * 3D inputs
  * Non-contiguous tensors
  * Dtype preservation
  * Epsilon sensitivity
  * Custom weights
  * In-place semantics
  * Large hidden sizes
  * Single-token cases

#### Performance

* **RMSNorm**: ~2.5× average speedup (up to 5.8×)
* **SiluAndMul**: ~1.8× average speedup
* No regressions observed vs. native implementation in any tested configuration.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/layers/layernorm.py [hunk 1]
```diff
@@ -14,12 +14,16 @@
...
 _is_cuda = current_platform.is_cuda()
 _is_npu = current_platform.is_npu()
+_is_musa = current_platform.is_musa()
 if _is_cuda:
     from sgl_kernel import fused_add_rmsnorm, rmsnorm
 
 if _is_npu:
     import torch_npu
 
+if _is_musa:
```
Comment (mickqian) [inline python/sglang/multimodal_gen/runtime/layers/layernorm.py:24]: could you try to do it with removing all these? thanks

File: test/registered/mthreads/diffusion/test_musa_rmsnorm.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (mickqian) [inline test/registered/mthreads/diffusion/test_musa_rmsnorm.py]: are these test files really activated?

Comment (yeahdongcn) [reply test/registered/mthreads/diffusion/test_musa_rmsnorm.py]: I originally planned to use these test files for the upcoming diffusion CI on MUSA. I’ll remove them from this PR and include them in the diffusion CI PR instead.

File: python/sglang/multimodal_gen/runtime/layers/layernorm.py (hunk 1 above)
Comment (yeahdongcn) [reply python/sglang/multimodal_gen/runtime/layers/layernorm.py:24]: I’m not entirely sure I understand the suggestion. `fused_add_rmsnorm` is required on MUSA to handle RMSNorm when a residual is present.

For the other ops, we can fall back to the built-in implementations in `torch.nn.functional`, but in this case, the fused kernel is necessary.

File: test/registered/mthreads/diffusion/test_musa_rmsnorm.py (hunk 2 above)
Comment (yeahdongcn) [reply test/registered/mthreads/diffusion/test_musa_rmsnorm.py]: Removed.

**Review by mickqian — APPROVED** [review]
